test/dynamo/test_misc.py

```diff
@@ -4697,6 +4698,27 @@ def test_remove_dead_code_with_exn_table_entries(self):
         self.assertEquals(tab[0].end, 4)
         self.assertEquals(tab[0].target, 6)
 
+    def test_unhandled_exception_in_dynamo(self):
+        # need to run in a separate process to properly test an unhandled
+        # exception raised in dynamo optimized code
+        code = """\
+import torch
+def f(a):
+    a += 1
+    raise RuntimeError("smoge")
+    return a
+
+opt_fn = torch.compile(backend="eager")(f)
+opt_fn(torch.ones(2))
+"""
+        proc = subprocess.run(
+            ["python3", "-c", code],
+            capture_output=True,
+        )
+        # process did not segfault, C assert fail (abort), etc.
+        self.assertGreater(proc.returncode, 0)
```

Comment:
Segfault would actually give you a non-zero code?

Reply:
Exits due to signals (segfault, abort, etc.) will have negative return code.

Reply:
I don't think that this is generally true?
This is not true on my Fedora install at least:
```
$ python foo.py 
Segmentation fault (core dumped)
$ echo $?
139
```

Reply:
This behavior is specific to python: https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.returncode

Reply:
We'll skip the test if not on POSIX

Reply:
Update: test changed to trigger segfault without spawning a subprocess. POSIX check no longer required.